src/Card.js

```diff
@@ -0,0 +1,126 @@
+import classNames from 'classnames';
+import React from 'react';
+import PropTypes from 'prop-types';
+import elementType from 'prop-types-extra/lib/elementType';
+
+import { createBootstrapComponent } from './ThemeProvider';
+import createWithBsPrefix from './utils/createWithBsPrefix';
+import CardContext from './CardContext';
+
+import CardImg from './CardImg';
+
+class Card extends React.Component {
+  static propTypes = {
+    /**
+     * @default 'card'
+     */
+    bsPrefix: PropTypes.string.isRequired,
+
+    /**
+     * Sets card background
+     *
+     * @type {('primary'|'secondary'|'success'|'danger'|'warning'|'info'|'dark'|'light')}
+     */
+    bg: PropTypes.string,
+
+    /**
+     * Sets card text color
+     *
+     * @type {('primary'|'secondary'|'success'|'danger'|'warning'|'info'|'dark'|'light'|'white'|'muted')}
+     */
+    text: PropTypes.string,
+
+    /**
+     * Sets card border color
+     *
+     * @type {('primary'|'secondary'|'success'|'danger'|'warning'|'info'|'dark'|'light')}
+     */
+    border: PropTypes.string,
+
+    /**
+     * When this prop is set, it creates a Card with a Card.Body inside
+     * passing the children directly to it
+     */
+    body: PropTypes.bool,
+
+    as: elementType
+  };
+
+  static defaultProps = {
+    as: 'div',
+    body: false
+  };
+
+  static getDerivedStateFromProps({ bsPrefix }) {
+    return {
+      cardContext: {
+        cardHeaderBsPrefix: `${bsPrefix}-header`
+      }
+    };
+  }
+
+  state = {};
+
+  render() {
+    const {
+      bsPrefix,
+      className,
+      as: Component,
+      bg,
+      text,
+      border,
+      body,
+      ...props
+    } = this.props;
+
+    const classes = classNames(
+      className,
+      bsPrefix,
+      bg && `bg-${bg}`,
+      text && `text-${text}`,
+      border && `border-${border}`
+    );
+
+    if (body) {
+      const { children, ...rest } = props;
+      return (
+        <Component className={classes} {...rest}>
+          <div className="card-body">{children}</div>
```

Comment:
you're missing the Provider here, plus i'd use Card.Body directly, even if you have to create above first.

I'd combine this into one branch as well:
```jsx
<CardContext.Provider value={this.state.cardContext}>
  <Component className={classes} {...props}>
    {body ? <CardBody>{children}</CardBody> : children}
  </Component>
</CardContext.Provider>
```